rs-matter/src/dm/subscriptions.rs

```diff
@@ -70,32 +70,56 @@ impl Subscription {
     }
 }
 
+struct SubscriptionData<const N: usize> {
```

Comment:
Supernit: most often that not when implementing the interior mutability pattern, the "inner" mutable structure is called like the parent one with a suffix "Inner". I.e., `SubscriptionInner`. However, this is not like a hard rule, and moreover - you have an extra member inside the `Subscription` struct which indeed should stay outside of `SubscriptionData`, so I guess `SubscriptionData` is fine as a name as well.

Reply:
Done, thanks!